apps/api/src/service/applicationService.ts

```diff
@@ -356,6 +357,25 @@ const applicationSvc = (db: PostgresDb) => ({
 			return failure(message, exception);
 		}
 	},
+
+	rejectApplication: async ({ applicationId }: { applicationId: number }): AsyncResult<ApplicationModel> => {
+		try {
+			const application = await db
+				.update(applications)
+				.set({ state: 'REJECTED', updated_at: sql`NOW()` })
+				.where(eq(applications.id, applicationId))
+				.returning();
+
+			if (!application[0]) throw new Error('Application record is undefined');
+
+			return success(application[0]);
+		} catch (exception) {
+			const message = `Error at rejectApplication with applicationId: ${applicationId}.`;
+			logger.error(message);
+			logger.error(exception);
+			return failure(message, exception);
+		}
+	},
```

Comment:
Related to discussion here: https://github.com/Pan-Canadian-Genome-Library/daco/pull/179#pullrequestreview-2607042602

The State Manager handles this DB update, so this service method won't be needed